src/components/FullscreenModal/Header/Header.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import { getColor, pxToRem } from '../../../utils/helpers';
+import { Button } from '../../Button';
+import {
+  ButtonColors,
+  ButtonSizes,
+  ButtonVariants,
+} from '../../Button/Button.enums';
+import { FlexContainer } from '../../FlexContainer';
+import { Icon } from '../../Icon';
+import { SSCIconNames } from '../../Icon/Icon.enums';
+import { Col, Container, Row } from '../../layout';
+import { H2, H3 } from '../../typography';
+import { useStickyHeader } from '../hooks/useStickyHeader';
+import { HeaderProps } from './Header.types';
+
+const BaseStickyHeader = styled.header`
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100%;
+  background-color: ${getColor('graphite5H')};
+  border-bottom: 1px solid ${getColor('graphiteHB')};
+`;
+
+const BaseHeader = styled.header`
+  padding: ${pxToRem(56, 0, 24)};
+`;
+
+const Header: React.FC<HeaderProps> = ({
+  children,
+  width,
+  offset,
+  modalRef,
+  handleClose,
+}) => {
+  const modalHeaderRef = useRef(null);
+  const { isFixed } = useStickyHeader(modalRef, modalHeaderRef);
+
+  return (
+    <>
+      {isFixed && (
```

Comment:
When it's fixed, why are both headers displayed?

Reply:
I tried to do this with just one header displayed at a time, but it's making a mess when the one should be displayed and the second one hidden. Because it will change the`modalRef. scrollHeight` and it usually caused an infinite loop when the second footer gets displayed and the condition in hook get change which causes the second footer gets hidden and again and again :)